### PR TITLE
vms ported to django 2.0 and python 3.6.3

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
### Description
Vms ported to django 2.0 and python 3.6.3

Fixes # https://github.com/systers/vms/issues/457

### Type of Change:
- Code

Code/Quality Assurance Only
- This change requires a documentation update (software upgrade on readme file)

 Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials

Code/Quality Assurance Only
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak 
Except for the issue https://github.com/systers/vms/issues/532 . Porting is complete. Please review.

File: vms/authentication/urls.py [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-
-urlpatterns = patterns('',
-    url(r'^$', views.index, name='index'),
-    url(r'^login/$', anonymous_required(auth_views.login), {'template_name': 'authentication/login.html'}, name='login_process'),
-    url(r'^logout/$', auth_views.logout, {'template_name': 'home/home.html'}, name='logout_process'),
-)
+app_name='authentication'
+urlpatterns = [
+    path('', views.index, name='index'),
+    path('login/', anonymous_required(auth_views.login), {'template_name': 'authentication/login.html'}, name='login_process'),
```
Comment (tonythomas01) [inline vms/authentication/urls.py]: This line is too long actually. later when we integrate pep8 checks, you will have to change it.

Comment (anjali-dhanuka) [reply vms/authentication/urls.py]: I have made the changes you suggested. Please review now.

Comment (tonythomas01) [reply vms/authentication/urls.py]: Looks good now. I think the next big thing would be to get this pass the `CI`

**Comment by anjali-dhanuka** [comment]
There are some issues in travis.yml:
1. DJANGO_SECRET_KEY must be used in place of SECRET_KEY
2.Selenium driver 2.53.1 has compatibility issues with mozilla firefox
latest version.
3. Even if we update selenium package geckodriver needs to be downloaded in
the pc.

**Comment by tonythomas01** [comment]
> 1. DJANGO_SECRET_KEY must be used in place of SECRET_KEY
I think you can update it in `settings.py` and in travis.yml ? 
For the pep8 (flake8) failings, I think we will have to run something like https://github.com/google/yapf on the entire codebase and make sure things are clean. God - the output is shameful. I will raise another ticket for that.

**Comment by anjali-dhanuka** [comment]
I previously updated the secret key in travis.yml but Tepasawani Pathak asked me to keep it the way it is.

**Comment by tonythomas01** [comment]
> I previously updated the secret key in travis.yml but Tepasawani Pathak asked me to keep it the way it is.
@tapasweni-pathak, what do you recommend as an alternative then ?

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 language: python
 python:
-  - "2.7"
+  - "3.6"
```
Comment (tapasweni-pathak) [inline .travis.yml]: You also need to update the commands at https://github.com/anjali-dhanuka/vms/blob/8ebafec862a960efcdd4d612a97116f1dfabe35c/.travis.yml#L17, `syncdb` will not work.

Comment (anjali-dhanuka) [reply .travis.yml]: @tapasweni-pathak Commands updated.
Please have a look now.

File: .travis.yml [hunk 3]
```diff
@@ -13,8 +13,9 @@ before_script:
   - export DJANGO_SECRET_KEY=foobarbaz
 script:
   - cd vms
-  - python manage.py syncdb --noinput
-  - python manage.py migrate --noinput --traceback --settings=vms.settings
+  - flake8 . 
+  - python3.6 -m manage.py syncdb --noinput
+  - python3.6 -m manage.py migrate --noinput --traceback --settings=vms.settings
```
Comment (tapasweni-pathak) [inline .travis.yml]: 👋 Is there any specific advantage of not using `makemigrations` and `migrate` but using -n and using `syncdb` instead?

Comment (tonythomas01) [reply .travis.yml]: it shoudl be `makemigrations` and `migrate` please. `syncdb` is deprecated.

Comment (tapasweni-pathak) [reply .travis.yml]: actually thanks for leaving this note real quick @tonythomas01 ! and keeping an 👀

Comment (tonythomas01) [reply .travis.yml]: even though, this `python3.6` can be replaced with just `python` right ?

Comment (anjali-dhanuka) [reply .travis.yml]: @tonythomas01 should I change it to makemigrations and migrate?

Comment (tonythomas01) [reply .travis.yml]: yes, please.

Comment (anjali-dhanuka) [reply .travis.yml]: @tonythomas01 Done! Please review.

File: vms/vms/settings.py [hunk 4]
```diff
@@ -70,15 +70,29 @@
...
+       'BACKEND': 'django.template.backends.django.DjangoTemplates',
+        'DIRS': [],
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'context_processors': [
+                'django.template.context_processors.debug',
+                'django.template.context_processors.request',
+                'django.contrib.auth.context_processors.auth',
+                'django.contrib.messages.context_processors.messages',
+                # 'django.contrib.auth.context_processors.auth',
```
Comment (tapasweni-pathak) [inline vms/vms/settings.py]: can you please delete them.

File: vms/vms/settings.py [hunk 5]
```diff
@@ -70,15 +70,29 @@
...
+                 #'django.core.context_processors.i18n',
+                # 'django.core.context_processors.media',
+                # 'django.core.context_processors.static',
+                # 'django.core.context_processors.tz',
+                # 'django.contrib.messages.context_processors.messages',
+                ],
+            },
+         },
+]
+
```
Comment (tapasweni-pathak) [inline vms/vms/settings.py]: Here you will find `TEMPLATE_DEBUG = DEBUG` which should be changed to `TEMPLATES[0]['OPTIONS']['debug'] = DEBUG`.

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak  Done with all your suggested changes. Please review.

File: requirements.txt [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 docopt==0.6.2
 flake8==3.4.1
 psycopg2==2.7.3
 PyYAML==3.11
 requests==2.7.0
-selenium==2.53.1
-phonenumbers==7.2.6
+selenium==3.8.0
+phonenumbers==8.8.8
 django-cities-light==3.2.0
```
Comment (tapasweni-pathak) [inline requirements.txt]: You would have to upgrade `django-cities-light` version to 3.3.0. What happens when you try to setup VMS in a new virtual environment?

Comment (anjali-dhanuka) [reply requirements.txt]: During setup in a virtual environment even with 3.3.0 version basically 2 types of errors are displayed:
In Django 2.0 'on_delete' is a necessary argument while defining fields of a model.
The old django.conf.urls.include() function is now to be imported from django.urls.

I have checked their repository they are already working on it https://github.com/yourlabs/django-cities-light/issues/166
Same is the case with django braces.

**Comment by anjali-dhanuka** [comment]
I fixed the errors manually. Should I push that on github too?

**Comment by tapasweni-pathak** [comment]
@anjali-dhanuka I don't think you would be able to push the changes of django-cities-light in VMS. As it is a separate package and we are its users, it is not useful to fixes and have the code in VMS, hope this answers the questions.

The useful approach to have this works would be do the changes and make VMS upgraded to use Django 1.11, I don't think it will have any issues as most packages are quite ready with adjusting themselves for 1.11 but not yet for 2.0, guess its been just two months.

**Comment by anjali-dhanuka** [comment]
We can port it to django 2.0 when these packages release their corresponding versions. So for now, should I start working on porting it to django 1.11 and python 3.6?

**Comment by tapasweni-pathak** [comment]
> We can port it to django 2.0 when these packages release their corresponding versions. 

Perfect.

> So for now, should I start working on porting it to django 1.11 and python 3.6?

Please, that would be great! thank you.

**Comment by tapasweni-pathak** [comment]
@anjali-dhanuka this needs a resolution of conflicts.

**Comment by anjali-dhanuka** [comment]
@tapasweni-pathak I have resolved the conflicts. Please review.

**Comment by anjali-dhanuka** [comment]
@tonythomas01 @tapasweni-pathak The django compatible version is relaesed for django cities light. See this https://github.com/yourlabs/django-cities-light/issues/166#issuecomment-353937410 . Plus I am done with all the changes u suggested. Please review.

**Comment by ajatprabha** [comment]
@anjali-dhanuka @tonythomas01 @tapasweni-pathak Hi there, I wanted to ask whether the port to Django 2.0 is completed? If not, I'd like to help port this.

**Comment by anjali-dhanuka** [comment]
@ajatprabha The porting has been done. The python packages it is using, their versions compatible with django 2.0 have not been released yet. Thus we have shifted to django 1.11.

**Comment by ajatprabha** [comment]
@anjali-dhanuka, thanks for the reply. I see you've got a PR ready for it in #558 but the Travis build is failing, I hope you get around it. 😊
One more thing, the previous build on v1.7.4 shows 42% code coverage, which is in pretty bad shape. Can you tell me what percentage of code coverage, the v1.11 port gives at present? Let me know if can be of any help in writing tests for the v1.11 port.

**Comment by tonythomas01** [comment]
Is this superseded by #678 ? If so, please close this PR.

**Comment by ExactlyMay** [comment]
There's a compatibility issue, so closing for now until a solution is found.

**Pull request closed without merging**